examples/datasets/front_3d_object_sampling/main.py

```diff
@@ -0,0 +1,154 @@
+import blenderproc as bproc
+import os
+import numpy as np
+import argparse
+import bpy
+
+from blenderproc.python.utility.CollisionUtility import CollisionUtility
+parser = argparse.ArgumentParser()
+parser.add_argument("front", help="Path to the 3D front file")
+parser.add_argument("future_folder", help="Path to the 3D Future Model folder.")
+parser.add_argument("front_3D_texture_path", help="Path to the 3D FRONT texture folder.")
+parser.add_argument('treed_obj_path', help="Path to the downloaded 3D object")
+parser.add_argument('output_dir', nargs='?', default="examples/datasets/front_3d_object_sampling/output", help="Path to where the final files, will be saved")
+args = parser.parse_args()
+
+if not os.path.exists(args.front) or not os.path.exists(args.future_folder) or not os.path.exists(args.shapenet_path):
+    raise Exception("One of the three folders does not exist!")
+
+bproc.init()
+mapping_file = bproc.utility.resolve_resource(os.path.join("front_3D", "3D_front_mapping.csv"))
+mapping = bproc.utility.LabelIdMapping.from_csv(mapping_file)
+
+# set the light bounces
+bproc.renderer.set_light_bounces(diffuse_bounces=200, glossy_bounces=200, max_bounces=200,
+                                  transmission_bounces=200, transparent_max_bounces=200)
+
+# load the front 3D objects
+room_objs = bproc.loader.load_front3d(
+    json_path=args.front,
+    future_model_path=args.future_folder,
+    front_3D_texture_path=args.front_3D_texture_path,
+    label_mapping=mapping
+)
+
+# define the camera intrinsics
+bproc.camera.set_resolution(512, 512)
+
+# Select the objects, where other objects should be sampled on
+left_objects = []
+for obj in room_objs:
+    if "table" in obj.get_name().lower() or "desk" in obj.get_name().lower():
+        left_objects.append(obj)
+
+for obj in left_objects:
+    # The loop starts with and UndoAfterExecution in order to restore the camera positions for each object
```

Comment:
This comment is a bit misleading, because it sounds like there exist already cam poses for each object and they are restored, however its more like the cam poses from the previous iteration are cleaned up. Also, the dropped objects are removed and the sliced up objects are restored.

Reply:
I agree